### DISCUSSION TITLE
loose calling window focus when closing a dialog box - Lazarus-ide/freepascal

### DISCUSSION THREAD

**Discussion by nicolasauthier** [elementary/triage#295] [Q&A] [answered]
Hi,
I am developing using the Lazarus IDE and free pascal (GTK2). I'm newly switching to eOS as a main developing platform.  
When I open a dialog box either with the IDE or with a compiled app (see screenshot enclosed) the calling greyed window never returned to normal after closing the dialog box.
 This does not append under windows or other linux distro.
Is this due to some specific theming things done for GTK in eOS 7 ?
Thanks for your help.

![screenshot](https://github.com/elementary/triage/assets/88929005/72ee76d2-dcb4-43d3-9a8a-c0b16ec3ba9c)

**Comment by lenemter** [comment]
This looks similar to https://github.com/elementary/gala/issues/1331 bug, which has been fixed. Have you fully updated your system?

**Comment by nicolasauthier** [comment]
Exactly. This is the same bug. Then, this is not fixed under eOS 7. Yes I think my version is up to date.
Thanks for answering.
Some advice ?
Also, there seemed this GTK app have no theme (IDE or the apps you can creaye with it). Any clues ? Can they share eOS theme ?
![image](https://github.com/elementary/triage/assets/88929005/57fd58c1-d074-479c-b1b4-c7d80a8ce8a4)

**Reply by lenemter** [reply]
Can you provide some example project because I cannot reproduce the issue in Lazarus IDE itself please?

**Comment by nicolasauthier** [comment]
Hi again,
Find enclose  the simplest app possible. There is a menu File with an Open and close menu item. Open does nothing else than opening the dialog box. Choose close or dismiss or Ok, then you return to the main form, always greyed...
Weird that you do not reproduce this behaviour with your conf. 
Thanks for helping.


[BasicMainWindow.zip](https://github.com/elementary/triage/files/12402136/BasicMainWindow.zip)

**Reply by lenemter** [reply]
That's weird, I cannot reproduce the issue here either. Can you execute `dpkg -s gala | grep Version` in the Terminal and paste it here, so I can get the installed version of the window manager, please?

**Comment by nicolasauthier** [comment]
Version: 7.1.1~r2034+pkg83~ubuntu7.1

I did install  Lazarus-ide (2.2.0) and FPC(3.2.2)  version using : 
sudo apt install fpc
sudo apt install fpc-source
sudo apt install lazarus

**Reply by lenemter** [reply, marked as answer]
There is actually a newer version of gala available, so can you fully update your system through AppCenter and check if is fixed please?

**Reply by nicolasauthier** [reply]
Great! This fix the bug.
Thanks a lot lenemter!

If I may,  just a last question. The gtk windows looks old fashion, as they missed the eOS theme.  Is there also a way to fix this ?
Merci beaucoup

**Reply by lenemter** [reply]
elementary OS doesn't officially support gtk2 because it's basically dead for 5 years already, that's why it looks this bad. There is https://github.com/Romchec/elementary-OS-gtk2-support, which adds gtk2 support, but it's community made and may not work as good as gtk3 stylesheet.

**Reply by nicolasauthier** [reply]
I have installed pantheon-tweaks and Arc theme.
My app looks much better.
Thanks
![Capture d'écran du 2023-08-22 13 39 18](https://github.com/elementary/triage/assets/88929005/c322886a-da2b-4833-bd94-9925f3f81408)
